backend/inspirehep/records/tasks.py

```diff
@@ -135,3 +142,17 @@ def regenerate_author_records_table_entries(uuids_to_regenerate):
         record.update_authors_records_table()
         record.update_self_citations()
         db.session.commit()
+
+
+def remove_duplicate_refs_from_record(matched_inspire_record, referenced_records_in_path, path):
+    affiliations_path = ".".join(path.split(".")[:-2])
+    affiliations = flatten_list(get_value(matched_inspire_record, affiliations_path, []))
```

Comment:
Can you change the naming, please? These are references, not affiliations :) 
Can we stick to the one version of extracting path to be consistent? So if you want get rid of `record.$ref` you do (imo it will be more readable)
```
referenced_records_in_parent_path = flatten_list(
                get_value(matched_inspire_record, path[: -len(".record.$ref")], [])
            )
```

Reply:
I changed the naming, but I'm using the ```".".join(path.split(".")[:-2])``` for a reason.
Some of the reference paths don't have ```".record.$ref"``` record in them.
for example:
```
'publication_info.parent_record.$ref'
'publication_info.journal_record.$ref'
'publication_info.conference_record.$ref'
```
So I wanted it to be dynamic.

Reply:
ok, I see. But that means that you take path parent, so maybe take the first element only?

Reply:
I would, but then the second problem is that some of the paths have 4 "steps" in them.😅 
For example:
```
'authors.affiliations.record.$ref'
'thesis_info.institutions.record.$ref'
```